### PR TITLE
Allow comments in domains file

### LINKED ISSUE #1333: Allow comments in domains file
This pull requests allows the domains file to have comments, prefixed with `#`. It strips everything after `#` character, and then does not process empty lines.

I.E.
```
# ProjectDiscovery
projectdiscovery.org # This is the main site
example.com # This is a demo website
```
Will be processed as
```
projectdiscovery.org
example.com
```

ref: https://github.com/projectdiscovery/subfinder/pull/1215#issue-2228678327

### PR DESCRIPTION
**Pull request by RedYetiDev**
This pull requests allows the domains file to have comments, prefixed with `#`. It strips everything after `#` character, and then does not process empty lines. Closes #1333.

I.E.
```
# ProjectDiscovery
projectdiscovery.org # This is the main site
example.com # This is a demo website
```
Will be processed as
```
projectdiscovery.org
example.com
```

### CODE REVIEW

**Comment by RedYetiDev** [comment]
If this PR is approved, I can squash if need be.

**Comment by GeorginaReeder** [comment]
Thanks for your contribution @RedYetiDev , we appreciate it! :)

File: v2/pkg/runner/runner.go [hunk 1]
```diff
@@ -119,9 +119,20 @@ func (r *Runner) EnumerateMultipleDomainsWithCtx(ctx context.Context, reader io.
 	scanner := bufio.NewScanner(reader)
 	ip, _ := regexp.Compile(`^([0-9\.]+$)`)
 	for scanner.Scan() {
-		domain, err := normalizeLowercase(scanner.Text())
+		domain := normalizeLowercase(scanner.Text())
```
Comment (dogancanbakir) [inline v2/pkg/runner/runner.go:122]: how about something like this?
```go
if stringsutil.HasSuffixAny(domain, "#", "//") {
	continue
}
```

**Review by dogancanbakir — CHANGES_REQUESTED** [review]

**Comment by dogancanbakir** [comment]
raised new pr https://github.com/projectdiscovery/subfinder/pull/1346

**Pull request closed without merging**